common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalBlockHandler.java

```diff
@@ -394,7 +394,7 @@ private class ManagedBufferIterator implements Iterator<ManagedBuffer> {
         size = reduceIdAndChunkIds.length;
         blockDataForIndexFn = index -> mergeManager.getMergedBlockData(msg.appId, shuffleId,
           shuffleMergeId, reduceIdAndChunkIds[index], reduceIdAndChunkIds[index + 1]);
-      } else if (blockId0Parts.length == 3 && blockId0Parts[0].equals("rdd")) {
+      } else if (blockId0Parts.length == 3 && "rdd".equals(blockId0Parts[0])) {
```

Comment:
In what scenario `blockId0Parts[0]` is null?

Reply:
Just defensive programming. It's possible that `blockId0Parts[0]` is null and no possible that `"rdd"` is null

Reply:
Unfortunately, it doesn't looks like this code has any improvement like you claimed, @dcoliversun .
If `blockID0Parts[0] == null`, it fails already at line 383 before reaches this line 397.